utils/node_local_test.py

```diff
@@ -377,6 +388,11 @@ def __init__(self, conf, test_class=None, valgrind=False):
     def __del__(self):
         if self.running:
             self.stop(None)
+        server_file = os.path.join(self.agent_dir, '.daos_server.active.yml')
```

Comment:
what is the purpose of this line? also why does it end up in agent_dir?

Reply:
This is a generic bug-fix that I've included.  Something (possibly the daos server?) copies the server file to this temp file, and it used to use the current directory but seems to now go in the agent directory, the result of which is the os.rmdir() on L396 is failing, only as this is inside a __del__ method it reports a Traceback but doesn't fail anywhere.

Reply:
The location of the active settings file was changed in this commit:
```
commit c7fbb4bdfd10df8d790819dddbb0df8ccc6cbcaf
Author: Tom Nabarro <tom.nabarro@intel.com>
Date:   Thu Apr 15 18:10:13 2021 +0100

    DAOS-7002 control: revamp server.go (#4929)

    The server main function has become unwieldy after bolting on so
    much functionality to what was once a simple entry point.
    Decompose into testable utility functions and improve the unit test
    coverage.

    Notable behaviour changes:
      - write read-only active config file to runtime directory, not /tmp/
      - validate server config only once during start-up
      - initialize netdetect library only once during start-up

    Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>
```
The file is now written to the server runtime directory (customisable through socket_dir server config file parameter), so the agent_dir must be pointing to the socket_dir directory.

Reply:
socket_dir is hard-coded to socket_dir: /tmp/dnt_sockets in nlt so you may want to check that.